src/actions/BulkSetRackNo.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace hipanel\modules\server\actions;
+
+use hipanel\actions\Action;
+use hipanel\actions\SmartUpdateAction;
+use Yii;
+use yii\helpers\ArrayHelper;
+use yii\web\NotFoundHttpException;
+
+class BulkSetRackNo extends SmartUpdateAction
+{
+    public function init(): void
+    {
+        $this->data = function (Action $action, array $data): array {
+            $result = [];
+            foreach ($data['models'] as $model) {
+                $result['models'][] = $this->collection->getModel()::fromOriginalModel($model);
+            }
+            if (!$result['models']) {
```

Comment:
```suggestion
            if (empty($result['models'])) {
```

Не делай лишние нотисы